### ISSUE TITLE
Wrong "Related page" generating in 1.9.3 (works good in 1.9.1!) from *.md files

### ISSUE THREAD

**Issue by Beszt** [doxygen/doxygen#9051]
**Describe and reproduce the bug**
When i try generate HTML output from *.MD files with /page header i have a some bugs. For Example my code in tests.md:

```
# Unit tests Requirements
\page test_requirements Unit tests Requirements
```

In 1.9.3 my "Related pages" link for this file generated like "Unit tests Requiremen" without "t" and "s" is in body of page! In 1.9.1 it's all ok.

**Comment by albert-github** [comment]
The problem with the `s` missing at the end of the word "Requirements" I can reproduce. In 1.9.1 I see:
![image](https://user-images.githubusercontent.com/5223533/150127230-879f507c-74af-42e7-b97d-d171b756546a.png)

and in 1.9.2 (and current master 1.9.4 (99482978eb791a6159c4e409c42ec95acbbd2c1d)) I see:
![image](https://user-images.githubusercontent.com/5223533/150127350-580425ec-0eda-410b-a3cc-7e6a7678b335.png)

The problem with the `t` I don't see, please indicate and also (so we are tallking about the same problem):
- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by albert-github** [comment]
Regarding the problem with the `s` I've just pushed a proposed patch, pull request #9053

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release)

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.4.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
